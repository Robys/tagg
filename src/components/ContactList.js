import { useState} from "react"
import {Redirect} from 'react-router-dom'
import {Paper,Avatar,List} from '@material-ui/core'
import AvatarGroup from '@material-ui/lab/AvatarGroup';
import {Skeleton} from '@material-ui/lab'

/** List of contacts of the user
 * 
 * @param {Object} user - current user detected by the system
 */

function ContactList ({user}){
    const [selected,SetSelected] = useState(null)

    return (
        <div>
            {user ? 
            <Paper className="dashboard-contacts">
                <List className="dashboard-contacts-list">
                    {user.contacts.length > 0 ? 
                    <AvatarGroup max={4} style={{marginLeft:"20px"}}>
                {user.contacts.map(contact =>
                    <Avatar key={contact.id} 
                    className="contact-photo" 
                    src={contact.picture}
                    onClick={e =>SetSelected(contact.id)}/>
                    )}
                    </AvatarGroup>

                    : 
                    <div>
                        <Skeleton variant="circle" width={40} height={40} animation="wave"/>

                    </div> }
                </List>

            </Paper>
            : <div style={{margin:"20px"}}>
            <Skeleton variant="circle" width={40} height={40} animation="wave"/>

        </div>}

            {selected !== null ? <Redirect to={`messages/${selected}`}/> : ""}

        </div>
    )
}

export default ContactList