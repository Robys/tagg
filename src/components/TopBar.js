import { useState } from "react";
import {LOGOUT} from '../api'
import {AppBar,
    Button,
    List,
    ListItem,
    ListItemText,
    Toolbar, 
    IconButton, 
    Drawer} from "@material-ui/core"
import MenuIcon from '@material-ui/icons/Menu';

/** The TopBar is placed in the top of the application, providing a menu
 * for the user to access other pages as the profile, contacts, etc.
 * 
 * @param {Object} user - logged user detected by the system
 */

function TopBar({user}){
    const [anchorEl, setAnchorEl] = useState(null);

    const handleClick = (event) => {
        setAnchorEl(event.currentTarget);
      };
    
      const handleClose = () => {
        setAnchorEl(null);
      };
   

    return (
        <AppBar position="static" style={{backgroundColor:"#FD507A"}}>
            <Toolbar>
                <IconButton edge="start" 
                color="inherit" aria-label="menu" onClick={handleClick}>
                    <MenuIcon />
                </IconButton>
            </Toolbar>

            {user ? 
            <Drawer anchor="left" open={anchorEl} onClose={handleClose}>
            <List>
                <ListItem>
                    <ListItemText primary={<a href="/dashboard">Home</a>} />
                </ListItem>
            
                <ListItem>
                    <ListItemText primary={<a href="/notifications">Notificações</a>} />
                </ListItem>

                <ListItem>
                    <ListItemText primary={<a href="/requests">Trocas</a>} />
                </ListItem>

                <ListItem>
                    <ListItemText primary={<a href="/contacts">Contatos</a>} />
                </ListItem>

                <ListItem>
                    <ListItemText primary={
                    <a href={`/profile/${user.id}`}>{user.firstName}</a>
                    } />
                </ListItem>
                {user.roles !== "USER" ?
                <ListItem>
                    <ListItemText primary={<a href="/tools"> Ferramentas </a> } />
                </ListItem>
                : ""}
        
                <ListItem onClick={e => {LOGOUT()}}>
                <Button href="/"
                variant="contained" color="secondary">
                    Sair
                </Button>
                    
                </ListItem>
            </List>
            </Drawer>
            : ""}
        </AppBar>
        
    )
}

export default TopBar