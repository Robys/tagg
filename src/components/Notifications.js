import { useState,useEffect} from 'react'
import { AppBar,Tabs,Tab} from '@material-ui/core';
import {GETNOTIFICATIONS } from '../api'
import {ReadedNotifications,RecentNotifications} from '../components/NotificationTabs'

function a11yProps(index) {
    return {
      id: `simple-tab-${index}`,
      'aria-controls': `simple-tabpanel-${index}`,
    };
  }

/** List of notifications from a user
 * 
 * @constructor
 * @param {Object} user - logged user detected by the system
 */


const Notifications = ({user}) => {
    const [data,SetData] = useState()

    const [value, setValue] = useState(0);
    const handleChange = (event, newValue) => {
        setValue(newValue);
      };

    useEffect(()=>{
        const getUserNotifications = async () =>{
            if(user){
                const response = await GETNOTIFICATIONS(user.id)
                SetData(response)
            }
        }

        getUserNotifications()

    },[])

    //console.log(data)

    return (
        <div>
            <AppBar position="static">
            <Tabs 
            style={{backgroundColor:"#FFFF",color:"#232323"}}
            value={value}
            variant="scrollable"
            scrollButtons="auto" 
            onChange={handleChange} >
                <Tab label="Recentes" {...a11yProps(0)} />
                <Tab label="Visualizadas" {...a11yProps(1)} />
            </Tabs>
            </AppBar>

            <RecentNotifications value={value} index={0}>
                {data}
            </RecentNotifications>
            <ReadedNotifications value={value} index={1}>
                {data}
                </ReadedNotifications>
        </div>
    )

}

/**
 * 
 * 
 */


export default Notifications