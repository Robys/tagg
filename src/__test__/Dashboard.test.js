import React from "react";
import {shallow} from '../enzyme'
import TopBar from '../components/TopBar'
import ContactList from '../components/ContactList'
import Notifications from '../components/Notifications'

describe('test context hook',()=>{

    const context = {
        id: "60edb9b0b291b127e48a8778",
        roles: "ADMIN",
        active: true,
        firstName: "Administrador",
        lastName: "-",
        email: "adm@tagg.com",
        location: "São Paulo",
        createdAt: "13/07/2021",
        picture: "undefined",
        gamecollection: [
            {
                id: "610429da8d57550d245bdc34",
                title: "Doom 2016",
                gender: "ação",
                platform: "playstation",
                status: "usado",
                value: "40",
                description: "teste",
                createdAt: "30/07/2021",
                cover: "https://res.cloudinary.com/dahwijw8w/image/upload/v1627662802/Tagg/pgyilwxsji2svlchrres.jpg"
            }
        ],
        contacts: [
            {
                id: "60edbd0ddd108f2380cad3e9",
                active: true,
                firstName: "Penelope",
                lastName: "Lopes",
                email: "pe@email.com",
                location: "São Paulo",
                picture: "https://res.cloudinary.com/dahwijw8w/image/upload/v1626193179/Tagg/jqxlkjzelp8x9byda1yn.jpg"
            },
            {
                id: "60f051ab648d18294c24c343",
                active: true,
                firstName: "William",
                lastName: "Pacifico",
                email: "william@email.com",
                location: "São Paulo",
                picture: "https://res.cloudinary.com/dahwijw8w/image/upload/v1626362293/Tagg/lwlhmp3g9vwq9xxynpia.jpg"
            }
        ]
    }
    
    test('rendering dashboard components with context function', ()=>{
        const container = shallow(
            <div>
                <TopBar user={context}/>
                <ContactList user={context}/>
                <Notifications user={context}/>
            </div>
        )
        expect(container).toMatchSnapshot();
    })


})