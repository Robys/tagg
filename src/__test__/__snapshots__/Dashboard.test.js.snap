// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test context hook rendering dashboard components with context function 1`] = `
<div>
  <TopBar
    user={
      Object {
        "active": true,
        "contacts": Array [
          Object {
            "active": true,
            "email": "pe@email.com",
            "firstName": "Penelope",
            "id": "60edbd0ddd108f2380cad3e9",
            "lastName": "Lopes",
            "location": "São Paulo",
            "picture": "https://res.cloudinary.com/dahwijw8w/image/upload/v1626193179/Tagg/jqxlkjzelp8x9byda1yn.jpg",
          },
          Object {
            "active": true,
            "email": "william@email.com",
            "firstName": "William",
            "id": "60f051ab648d18294c24c343",
            "lastName": "Pacifico",
            "location": "São Paulo",
            "picture": "https://res.cloudinary.com/dahwijw8w/image/upload/v1626362293/Tagg/lwlhmp3g9vwq9xxynpia.jpg",
          },
        ],
        "createdAt": "13/07/2021",
        "email": "adm@tagg.com",
        "firstName": "Administrador",
        "gamecollection": Array [
          Object {
            "cover": "https://res.cloudinary.com/dahwijw8w/image/upload/v1627662802/Tagg/pgyilwxsji2svlchrres.jpg",
            "createdAt": "30/07/2021",
            "description": "teste",
            "gender": "ação",
            "id": "610429da8d57550d245bdc34",
            "platform": "playstation",
            "status": "usado",
            "title": "Doom 2016",
            "value": "40",
          },
        ],
        "id": "60edb9b0b291b127e48a8778",
        "lastName": "-",
        "location": "São Paulo",
        "picture": "undefined",
        "roles": "ADMIN",
      }
    }
  />
  <ContactList
    user={
      Object {
        "active": true,
        "contacts": Array [
          Object {
            "active": true,
            "email": "pe@email.com",
            "firstName": "Penelope",
            "id": "60edbd0ddd108f2380cad3e9",
            "lastName": "Lopes",
            "location": "São Paulo",
            "picture": "https://res.cloudinary.com/dahwijw8w/image/upload/v1626193179/Tagg/jqxlkjzelp8x9byda1yn.jpg",
          },
          Object {
            "active": true,
            "email": "william@email.com",
            "firstName": "William",
            "id": "60f051ab648d18294c24c343",
            "lastName": "Pacifico",
            "location": "São Paulo",
            "picture": "https://res.cloudinary.com/dahwijw8w/image/upload/v1626362293/Tagg/lwlhmp3g9vwq9xxynpia.jpg",
          },
        ],
        "createdAt": "13/07/2021",
        "email": "adm@tagg.com",
        "firstName": "Administrador",
        "gamecollection": Array [
          Object {
            "cover": "https://res.cloudinary.com/dahwijw8w/image/upload/v1627662802/Tagg/pgyilwxsji2svlchrres.jpg",
            "createdAt": "30/07/2021",
            "description": "teste",
            "gender": "ação",
            "id": "610429da8d57550d245bdc34",
            "platform": "playstation",
            "status": "usado",
            "title": "Doom 2016",
            "value": "40",
          },
        ],
        "id": "60edb9b0b291b127e48a8778",
        "lastName": "-",
        "location": "São Paulo",
        "picture": "undefined",
        "roles": "ADMIN",
      }
    }
  />
  <Notifications
    user={
      Object {
        "active": true,
        "contacts": Array [
          Object {
            "active": true,
            "email": "pe@email.com",
            "firstName": "Penelope",
            "id": "60edbd0ddd108f2380cad3e9",
            "lastName": "Lopes",
            "location": "São Paulo",
            "picture": "https://res.cloudinary.com/dahwijw8w/image/upload/v1626193179/Tagg/jqxlkjzelp8x9byda1yn.jpg",
          },
          Object {
            "active": true,
            "email": "william@email.com",
            "firstName": "William",
            "id": "60f051ab648d18294c24c343",
            "lastName": "Pacifico",
            "location": "São Paulo",
            "picture": "https://res.cloudinary.com/dahwijw8w/image/upload/v1626362293/Tagg/lwlhmp3g9vwq9xxynpia.jpg",
          },
        ],
        "createdAt": "13/07/2021",
        "email": "adm@tagg.com",
        "firstName": "Administrador",
        "gamecollection": Array [
          Object {
            "cover": "https://res.cloudinary.com/dahwijw8w/image/upload/v1627662802/Tagg/pgyilwxsji2svlchrres.jpg",
            "createdAt": "30/07/2021",
            "description": "teste",
            "gender": "ação",
            "id": "610429da8d57550d245bdc34",
            "platform": "playstation",
            "status": "usado",
            "title": "Doom 2016",
            "value": "40",
          },
        ],
        "id": "60edb9b0b291b127e48a8778",
        "lastName": "-",
        "location": "São Paulo",
        "picture": "undefined",
        "roles": "ADMIN",
      }
    }
  />
</div>
`;
